resolver/custom_dns_resolver.go

```diff
@@ -85,12 +92,22 @@ func (r *CustomDNSResolver) processRequest(request *model.Request) *model.Respon
 	domain := util.ExtractDomain(question)
 
 	for len(domain) > 0 {
-		ips, found := r.mapping[domain]
+		entries, found := r.mapping[domain]
 		if found {
-			for _, ip := range ips {
-				if isSupportedType(ip, question) {
-					rr, _ := util.CreateAnswerFromQuestion(question, ip, r.cfg.CustomTTL.SecondsU32())
-					response.Answer = append(response.Answer, rr)
+			for _, entry := range entries {
+				switch v := entry.(type) {
+				case *dns.A:
+					ip := net.ParseIP(v.A.String())
+					result := r.processIP(ip, question)
```

Comment:
Sorry if it wasn't clear but I mean directly passing the IP without going through a string:
```suggestion
					result := r.processIP(v.A, question)
```